internal/core/services/room_manager/room_manager.go

```diff
@@ -132,3 +132,33 @@ func (m *RoomManager) UpdateRoom(ctx context.Context, gameRoom *game_room.GameRo
 
 	return nil
 }
+
+// ListRoomsWithDeletionPriority returns a specified number of rooms, following
+// the priority of it being deleted (which will be introduced later). This
+// function can return less rooms than the `amount` since it might not have
+// enough rooms on the scheduler.
+func (m *RoomManager) ListRoomsWithDeletionPriority(ctx context.Context, schedulerName string, amount int) ([]*game_room.GameRoom, error) {
+	// TODO(gabrielcorado): implement the priority. for now, we're list all
+	// rooms and taking the necessary "amount".
+	schedulerRoomsIDs, err := m.roomStorage.GetAllRoomIDs(ctx, schedulerName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list scheduler rooms: %w", err)
+	}
+
+	var result []*game_room.GameRoom
+	for _, roomID := range schedulerRoomsIDs {
+		room, err := m.roomStorage.GetRoom(ctx, schedulerName, roomID)
+		if err != nil {
+			// TODO(gabrielcorado): should we fail the entire process because of
+			// a room missing?
```

Comment:
In what case we will have a missing room with an existent roomID from the scheduler?

Reply:
No, because we list them all, and they start getting more of their details.

My point here was more related to if this happens, we have an inconsistent state. In this case, what should we do:
* recover somehow? let's say clean the room with error from the list;
* add into a metric and then just measure those inconsistencies (seems the best first version here, altho I don't want to leave too many inconsistencies and deal with them later);